modules/alloy/src/main/ts/ephox/alloy/behaviour/blocking/BlockingApis.ts

```diff
@@ -0,0 +1,80 @@
+import { Optional, Optionals } from '@ephox/katamari';
+import { Attribute } from '@ephox/sugar';
+import * as BehaviourTypes from '../common/BehaviourTypes';
+import * as Behaviour from '../../api/behaviour/Behaviour';
+import { Keying } from '../../api/behaviour/Keying';
+import { Focusing } from '../../api/behaviour/Focusing';
+import { Replacing } from '../../api/behaviour/Replacing';
+import { AlloySpec } from '../../api/component/SpecTypes';
+import { AlloyComponent } from '../../api/component/ComponentApi';
+import * as GuiFactory from '../../api/component/GuiFactory';
+import { BlockingConfig } from './BlockingSchema';
+import { BlockingState } from './BlockingState';
+
+export interface BlockingApis extends BehaviourTypes.BehaviourApisRecord<BlockingConfig, BlockingState> {
+  block: typeof block;
+  unblock: typeof unblock;
+}
+
+// Mark this component as busy, or blocked.
+const block = (
+  component: AlloyComponent,
+  config: BlockingConfig,
+  state: BlockingState,
+  // This works in conjunction with the 'getRoot' function in the config. To
+  // attach a blocker component to the dom, ensure that 'getRoot' returns a
+  // component, and this function returns the specification of the component to
+  // attach.
+  getBusySpec?: (behaviour: Behaviour.AlloyBehaviourRecord) => AlloySpec
+) => {
+  if (!state.getBlocked()) {
+    config.onBlock(component);
+  }
+
+  state.setBlocked(true);
+
+  Attribute.set(component.element, 'aria-busy', true);
+
+  let hasNewBlocker = false;
+  Optionals.lift2(Optional.from(getBusySpec), config.getRoot(component), (getBusySpec, root) => {
```

Comment:
So it seems intentional that `getRoot` can return None. In what situation would you want `block` to silently fail (clearBlocker) because `getRoot` is none?

Reply:
I wrote this in an earlier comment, but my idea was to base things off of the Invalidating behaviour, and give people the option whether they wanted to let us create the blocker element or whether they wanted to provide their own visual indication to the users just using the event callbacks

Reply:
Fixed